dvc/odb/config.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from typing import TYPE_CHECKING
+
+from voluptuous import MultipleInvalid
+
+from dvc.exceptions import DvcException
+
+if TYPE_CHECKING:
+    from dvc.fs.base import BaseFileSystem
+    from dvc.path_info import PathInfo
+
+logger = logging.getLogger(__name__)
+
+
+CONFIG_FILENAME = "dvc.odb.yaml"
+
+
+class ODBConfigFormatError(DvcException):
+    pass
+
+
+def load_config(path_info: "PathInfo", fs: "BaseFileSystem",) -> dict:
+    from dvc.utils.serialize import load_yaml
+
+    from .versions import EMPTY_VERSION_INFO
+
+    logger.debug("Loading ODB config '%s'", path_info)
+    if fs.exists(path_info):
+        data = load_yaml(path_info, fs=fs)
+    else:
+        data = EMPTY_VERSION_INFO
+
+    _validate_version(data)
+    return data
+
+
+def dump_config(
+    config: dict, path_info: "PathInfo", fs: "BaseFileSystem",
+):
+    from tempfile import NamedTemporaryFile
+
+    from dvc.utils.serialize import modify_yaml
+
+    logger.debug("Writing ODB config '%s'", path_info)
+    if not fs.exists(path_info.parent):
+        fs.makedirs(path_info.parent)
+
+    with NamedTemporaryFile() as tmp:
+        with modify_yaml(tmp.name) as data:
+            data.update(config)
+        tmp.seek(0)
+        fs.upload_fobj(tmp, path_info)
```

Comment:
This seems like a good use case for `MemoryFileSystem`, but for some reason the fsspec memory implementation of `fs.open()` does not play nicely with the edit-in place `modify_yaml` functionality and I didn't think it was worth digging into the issue at this point.